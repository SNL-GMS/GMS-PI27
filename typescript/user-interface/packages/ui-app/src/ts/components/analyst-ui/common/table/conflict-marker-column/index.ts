export type {
  ConflictMarkerCellRendererParams,
  ConflictRow
} from './conflict-marker-cell-renderer';
export { getConflictColumnDef } from './conflict-marker-column-def';
