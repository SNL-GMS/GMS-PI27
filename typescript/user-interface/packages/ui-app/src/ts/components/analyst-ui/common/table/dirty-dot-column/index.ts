export type { DirtyDotCellRendererParams, UnsavedChangesRow } from './dirty-dot-cell-renderer';
export { getDirtyDotColumnDef } from './dirty-dot-column-def';
