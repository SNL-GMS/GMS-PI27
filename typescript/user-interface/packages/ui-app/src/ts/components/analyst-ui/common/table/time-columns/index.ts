export { getTimeColumnDef } from './time-column-def';
export { getTimeUncertaintyColumnDef } from './time-uncertainty-column-def';
export type { TimeRow } from './types';
