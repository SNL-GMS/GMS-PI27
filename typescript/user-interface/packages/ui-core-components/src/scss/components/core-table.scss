/// Blueprint Table
.core-table__container {
  .bp5-table-header .bp5-table-column-name {
    height: 100%;
  }
  .bp5-table-header .bp5-table-header-content {
    position: absolute;
    margin-left: 24px;
    top: 0;
    display: inline-flex;
  }
  .core-table__column-header {
    display: flex;
    align-items: center;
    width: calc(100% - 25px);
    & .bp5-icon.bp5-intent-primary {
      color: var(--core-main, var(--gms-main, $core-main));
    }
    .force-right {
      margin-left: auto;
      margin-right: 0.5em;
    }
  }

  /// Tooltips
  .bp5-table-cell .bp5-popover-target {
    width: 100%;
  }
}
.core-table__column-header--default {
  font-size: 16px;
  font-weight: 500;
  letter-spacing: 1px;
}
