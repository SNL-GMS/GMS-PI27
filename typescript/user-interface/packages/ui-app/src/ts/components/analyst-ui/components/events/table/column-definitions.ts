import { EventsColumn } from '@gms/ui-state';

import { getConflictColumnDef } from '~analyst-ui/common/table/conflict-marker-column';
import { getDirtyDotColumnDef } from '~analyst-ui/common/table/dirty-dot-column';
import {
  getTimeColumnDef,
  getTimeUncertaintyColumnDef
} from '~analyst-ui/common/table/time-columns';

import type { EventRow } from '../types';
import { eventColumnDisplayStrings } from '../types';
import { activeAnalystsColumnDef } from './columns/active-analysts';
import { confidenceSemiMajorColumnDef } from './columns/confidence-semi-major';
import { confidenceSemiMajorTrendColumnDef } from './columns/confidence-semi-major-trend';
import { confidenceSemiMinorColumnDef } from './columns/confidence-semi-minor';
import { coverageSemiMajorColumnDef } from './columns/coverage-semi-major';
import { coverageSemiMajorTrendColumnDef } from './columns/coverage-semi-major-trend';
import { coverageSemiMinorColumnDef } from './columns/coverage-semi-minor';
import { deletedColumnDef } from './columns/deleted';
import { depthColumnDef } from './columns/depth';
import { depthUncertaintyColumnDef } from './columns/depth-uncertainty';
import { latitudeColumnDef } from './columns/latitude';
import { longitudeColumnDef } from './columns/longitude';
import { mbColumnDef } from './columns/mb';
import { mlColumnDef } from './columns/ml';
import { msColumnDef } from './columns/ms';
import { numberAssociatedColumnDef } from './columns/number-associated';
import { numberDefiningColumnDef } from './columns/number-defining';
import { observationsStandardDeviationColumnDef } from './columns/observerations-standard-dev';
import { preferredColumnDef } from './columns/preferred';
import { regionColumnDef } from './columns/region';
import { rejectedColumnDef } from './columns/rejected';
import { statusColumnDef } from './columns/status';

/**
 * @returns List of column definition objects for the Events Table
 */
export const getEventsTableColumnDefs = () => {
  return [
    getDirtyDotColumnDef<EventRow>(
      eventColumnDisplayStrings.get(EventsColumn.unsavedChanges),
      EventsColumn.unsavedChanges,
      eventColumnDisplayStrings.get(EventsColumn.unsavedChanges)
    ),
    getConflictColumnDef<EventRow>(
      eventColumnDisplayStrings.get(EventsColumn.conflict),
      EventsColumn.conflict,
      eventColumnDisplayStrings.get(EventsColumn.conflict)
    ),
    getTimeColumnDef(eventColumnDisplayStrings.get(EventsColumn.time), EventsColumn.time, 'Time'),
    getTimeUncertaintyColumnDef<EventRow>(
      eventColumnDisplayStrings.get(EventsColumn.timeUncertainty),
      EventsColumn.timeUncertainty,
      'Event time standard deviation'
    ),
    latitudeColumnDef,
    longitudeColumnDef,
    depthColumnDef,
    depthUncertaintyColumnDef,
    mbColumnDef,
    msColumnDef,
    mlColumnDef,
    numberAssociatedColumnDef,
    numberDefiningColumnDef,
    observationsStandardDeviationColumnDef,
    coverageSemiMajorColumnDef,
    coverageSemiMinorColumnDef,
    coverageSemiMajorTrendColumnDef,
    confidenceSemiMajorColumnDef,
    confidenceSemiMinorColumnDef,
    confidenceSemiMajorTrendColumnDef,
    regionColumnDef,
    activeAnalystsColumnDef,
    preferredColumnDef,
    statusColumnDef,
    rejectedColumnDef,
    deletedColumnDef
  ];
};
