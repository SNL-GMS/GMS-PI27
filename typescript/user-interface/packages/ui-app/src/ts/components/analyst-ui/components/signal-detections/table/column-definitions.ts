import type { ColumnDefinition } from '@gms/ui-core-components';
import type { ArrivalTime } from '@gms/ui-state';
import { SignalDetectionColumn } from '@gms/ui-state';
import type { ICellRendererParams, IHeaderParams } from 'ag-grid-community';

import { getConflictColumnDef } from '~analyst-ui/common/table/conflict-marker-column';
import { getDirtyDotColumnDef } from '~analyst-ui/common/table/dirty-dot-column';
import {
  getTimeColumnDef,
  getTimeUncertaintyColumnDef
} from '~analyst-ui/common/table/time-columns';

import type { SignalDetectionRow } from '../types';
import { signalDetectionColumnDisplayStrings } from '../types';
import { amplitudeColumnDef } from './columns/amplitude';
import { assocStatusColumnDef } from './columns/assoc-status';
import { azimuthColumnDef } from './columns/azimuth';
import { azimuthStdDevColumnDef } from './columns/azimuth-std-dev';
import { channelColumnDef } from './columns/channel';
import { deletedColumnDef } from './columns/deleted';
import { emergenceAngleColumnDef } from './columns/emergence-angle';
import { longPeriodFirstMotionColumnDef } from './columns/long-period-first-motion';
import { periodColumnDef } from './columns/period';
import { phaseColumnDef } from './columns/phase';
import { phaseConfidenceColumnDef } from './columns/phase-confidence';
import { rectilinearityColumnDef } from './columns/rectilinearity';
import { shortPeriodFirstMotionColumnDef } from './columns/short-period-first-motion';
import { slownessColumnDef } from './columns/slowness';
import { slownessStandardDeviationColumnDef } from './columns/slowness-std-dev';
import { sNRColumnDef } from './columns/snr';
import { stationColumnDef } from './columns/station';

/**
 * @returns List of column definition objects for the Signal Detection Table
 */
export const getSignalDetectionTableColumnDefs = () => {
  /** Extend the default column definition to make it editable */
  const editableTimeColumnDef: ColumnDefinition<
    SignalDetectionRow,
    unknown,
    ArrivalTime,
    ICellRendererParams,
    IHeaderParams
  > = {
    ...getTimeColumnDef<SignalDetectionRow>(
      signalDetectionColumnDisplayStrings.get(SignalDetectionColumn.time),
      SignalDetectionColumn.time,
      'Arrival time'
    ),
    editable: params => params.data.deleted !== 'True',
    cellClass: 'ag-cell-is-editable'
  };

  /** Extend the default column definition to make it editable */
  const editableTimeUncertaintyColumnDef: ColumnDefinition<
    SignalDetectionRow,
    unknown,
    ArrivalTime,
    ICellRendererParams,
    IHeaderParams
  > = {
    ...getTimeUncertaintyColumnDef<SignalDetectionRow>(
      signalDetectionColumnDisplayStrings.get(SignalDetectionColumn.timeStandardDeviation),
      SignalDetectionColumn.timeStandardDeviation,
      'Arrival time standard deviation'
    ),
    editable: params => params.data.deleted !== 'True',
    cellClass: 'ag-cell-is-editable'
  };

  return [
    getDirtyDotColumnDef(
      signalDetectionColumnDisplayStrings.get(
        SignalDetectionColumn.unsavedChanges,
        'Unsaved changes'
      ),
      SignalDetectionColumn.unsavedChanges,
      signalDetectionColumnDisplayStrings.get(
        SignalDetectionColumn.unsavedChanges,
        'Unsaved changes'
      )
    ),
    assocStatusColumnDef,
    getConflictColumnDef(
      signalDetectionColumnDisplayStrings.get(SignalDetectionColumn.conflict, 'Conflict'),
      SignalDetectionColumn.conflict,
      signalDetectionColumnDisplayStrings.get(SignalDetectionColumn.conflict, 'Conflict')
    ),
    stationColumnDef,
    channelColumnDef,
    phaseColumnDef,
    phaseConfidenceColumnDef,
    editableTimeColumnDef,
    editableTimeUncertaintyColumnDef,
    azimuthColumnDef,
    azimuthStdDevColumnDef,
    slownessColumnDef,
    slownessStandardDeviationColumnDef,
    amplitudeColumnDef,
    periodColumnDef,
    sNRColumnDef,
    rectilinearityColumnDef,
    emergenceAngleColumnDef,
    shortPeriodFirstMotionColumnDef,
    longPeriodFirstMotionColumnDef,
    deletedColumnDef
  ];
};
