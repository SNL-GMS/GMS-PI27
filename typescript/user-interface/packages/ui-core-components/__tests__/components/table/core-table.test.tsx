/* eslint-disable react/jsx-props-no-spreading */
import { render } from '@testing-library/react';
import * as React from 'react';
import { act } from 'react-test-renderer';

import type { CoreTableProps } from '../../../src/ts/components/table/core-table';
import { CoreTable } from '../../../src/ts/components/table/core-table';
import type { ColumnDefinition, Row } from '../../../src/ts/ui-core-components';

// set up window alert and open so we don't see errors
(window as any).alert = jest.fn();
(window as any).open = jest.fn();

jest.mock('ag-grid-react', () => {
  const actual = jest.requireActual('ag-grid-react');

  function AgGridReact(props: any) {
    // eslint-disable-next-line react/destructuring-assignment
    props.onGridReady({
      api: {
        getModel: jest.fn(() => {
          return { getType: jest.fn(() => 'infinite') };
        }),
        forEachNode: jest.fn(fn => fn({})),
        refreshCells: jest.fn(),
        destroy: jest.fn(),
        setDatasource: jest.fn(),
        purgeInfiniteCache: jest.fn(),
        getRenderedNodes: jest.fn(() => {
          return [{}];
        })
      },
      columnApi: {}
    });
    // eslint-disable-next-line react/destructuring-assignment
    props.onColumnMoved({});
    return <div />;
  }

  return { ...actual, AgGridReact };
});

interface RowType extends Row {
  versionId: number;
  color: string;
  category: string;
  type: string;
  startTime: number;
  endTime: number;
  Ids: number[];
  rationale: string;
}

const generateTableRows = () => {
  const rows: RowType[] = [
    {
      id: '1',
      versionId: 1,
      color: 'RED',
      category: 'normal',
      type: 'info',
      startTime: 1000,
      endTime: 2000,
      Ids: [1, 2],
      rationale: 'user'
    },
    {
      id: '2',
      versionId: 1,
      color: 'GREEN',
      category: 'normal',
      type: 'info',
      startTime: 2000,
      endTime: 3000,
      Ids: [1, 2],
      rationale: 'user'
    }
  ];
  rows[0]['first-in-table'] = true;
  return rows;
};

const columns: any[] = [
  {
    headerName: 'test',
    field: 'test',
    headerTooltip: 'test',
    width: 10
  },
  {
    headerName: 'test2',
    field: 'test2',
    headerTooltip: 'test2',
    width: 10,
    sort: 'asc'
  }
];

const columnDefs: ColumnDefinition<RowType, unknown, unknown, unknown, unknown>[] = [...columns];
describe('Core Table', () => {
  const onRowClicked = jest.fn();
  const onCellClicked = jest.fn();
  const onCellContextMenu = jest.fn();
  const onRowSelected = jest.fn();
  const onColumnMoved = jest.fn();
  const tableProps: CoreTableProps<RowType, any> = {
    defaultColDef: undefined,
    onRowClicked,
    onCellClicked,
    onCellContextMenu,
    onRowSelected,
    onColumnMoved,
    columnDefs: columnDefs as any,
    rowData: generateTableRows(),
    debug: true
  };

  it('is exported', () => {
    expect(CoreTable).toBeDefined();
  });
  it('Renders', async () => {
    let result;
    await act(async () => {
      // eslint-disable-next-line @typescript-eslint/await-thenable
      result = await render(<CoreTable {...tableProps} />);
    });
    expect(result.container).toMatchSnapshot();
  });

  it('can handle rerenders with new props', () => {
    const result = render(<CoreTable {...tableProps} />);
    columnDefs.push({
      headerName: 'test3',
      field: 'test3',
      headerTooltip: 'test3',
      width: 10
    });
    result.rerender(<CoreTable columnDefs={columnDefs} />);
    expect(result.container).toMatchSnapshot();
  });
});
