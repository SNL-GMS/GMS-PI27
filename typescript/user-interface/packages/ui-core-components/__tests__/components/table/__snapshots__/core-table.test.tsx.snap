// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Core Table Renders 1`] = `
<div>
  <div />
</div>
`;

exports[`Core Table can handle rerenders with new props 1`] = `
<div>
  <div />
</div>
`;
